estate_account/models/estate_property.py

```diff
@@ -0,0 +1,128 @@
+
+    # trying to understand sales invoice creation line by line
+    def _create_invoices(self, grouped=False, date=None):
+        if not self.env['account.move'].has_access('create'):
+            try:
+                self.check_access('write')
+            except AccessError:
+                return self.env['account.move']
+
+        invoice_vals_list = []
+        for invoice_item_sequence, record in enumerate(self):
+            if record.state != 'sold':
+                raise UserError('You can only invoice sold properties.')
+
+            if record.buyer_id.lang:
+                record = record.with_context(lang=record.buyer_id.lang)
+
+            invoice_vals = record._prepare_invoice()
+            invoice_vals['invoice_line_ids'].append(
+                fields.Command.create(record._prepare_invoice_line(sequence=invoice_item_sequence))
+            )
+            invoice_vals_list.append(invoice_vals)
+
+        if not grouped:
+            new_invoice_vals_list = []
+            invoice_grouping_keys = self._get_invoice_grouping_keys()
+            invoice_vals_list = sorted(
+                invoice_vals_list,
+                key=lambda x: [
+                    x.get(grouping_key) for grouping_key in invoice_grouping_keys
+                ]
+            )
+
+            for _grouping_keys, invoices in groupby(invoice_vals_list,
+                                                    key=lambda x: (x.get(grouping_key) for grouping_key in
+                                                                   invoice_grouping_keys)):
+                origins = set()
+                refs = set()
+                ref_invoice_vals = None
+                for invoice_vals in invoices:
+                    if not ref_invoice_vals:
+                        ref_invoice_vals = invoice_vals
+                    else:
+                        ref_invoice_vals['invoice_line_ids'] += invoice_vals['invoice_line_ids']
+                    origins.add(invoice_vals['invoice_origin'])
+                    refs.add(invoice_vals['ref'])
+
+                ref_invoice_vals.update({
+                    'ref': ', '.join(refs)[:2000],
+                    'invoice_origin': ', '.join(origins)
+                })
+                new_invoice_vals_list.append(ref_invoice_vals)
+            invoice_vals_list = new_invoice_vals_list
+
+            return self.env['account.move'].sudo().with_context(default_move_type='out_invoice').create(
+                invoice_vals_list)
+
+    def action_view_invoice(self):
+        action = self.env['ir.actions.actions']._for_xml_id('account.action_move_out_invoice_type')
+        form_view = [(self.env.ref('account.view_move_form').id, 'form')]
+        if 'views' in action:
+            action['views'] = form_view + [(state, view) for state, view in action['views'] if view != 'form']
+        else:
+            action['views'] = form_view
+        action['res_id'] = self.invoice_id.id
+
+        return action
```

Comment:
This is also too complicated with what is needed. In python, you can return a dict to communicate an action (unless the method is called from a simple other python method). Your case should not need more than this (you'll probably want to adapt it to make it work):
```suggestion
        return {
            'name': 'Invoice',
            'view_mode': 'form',
            'res_model': 'account.move',
            'type': 'ir.actions.act_window',
            'res_id': self.invoice_id.id,
        }
```